examples/TriangulationLOSTExample.cpp

```diff
@@ -131,23 +132,23 @@ int main(int argc, char* argv[]) {
         AddNoiseToMeasurements(measurements, measurementSigma);
 
     auto lostStart = std::chrono::high_resolution_clock::now();
-    boost::optional<Point3> estimateLOST = triangulatePoint3<Cal3_S2>(
+    auto estimateLOST = triangulatePoint3<Cal3_S2>(
         cameras, noisyMeasurements, rank_tol, false, measurementNoise, true);
     durationLOST += std::chrono::high_resolution_clock::now() - lostStart;
 
     auto dltStart = std::chrono::high_resolution_clock::now();
-    boost::optional<Point3> estimateDLT = triangulatePoint3<Cal3_S2>(
+    auto estimateDLT = triangulatePoint3<Cal3_S2>(
         cameras, noisyMeasurements, rank_tol, false, measurementNoise, false);
     durationDLT += std::chrono::high_resolution_clock::now() - dltStart;
 
     auto dltOptStart = std::chrono::high_resolution_clock::now();
-    boost::optional<Point3> estimateDLTOpt = triangulatePoint3<Cal3_S2>(
+    auto estimateDLTOpt = triangulatePoint3<Cal3_S2>(
         cameras, noisyMeasurements, rank_tol, true, measurementNoise, false);
     durationDLTOpt += std::chrono::high_resolution_clock::now() - dltOptStart;
 
-    errorsLOST.row(i) = *estimateLOST - landmark;
-    errorsDLT.row(i) = *estimateDLT - landmark;
-    errorsDLTOpt.row(i) = *estimateDLTOpt - landmark;
+    errorsLOST.row(i) = estimateLOST - landmark;
```

Comment:
I don't understand - we still have to de-reference std::optionals as well, no?

Reply:
triangulatePoint3 returns a Point3 object. I don't know why we used optionals there. It should up when I replaced all boost::optionals with `auto` so I fixed it.

Reply:
@akshay-krishnan should LOST triangulation also not return none when it fails?

Reply:
*Likely unrelated to this PR*

The DLT code throws a `TriangulationUnderconstrainedException`, but LOST does not. Is that what you mean? I dont see any other differences. 
This difference comes from which API we use to solve the linear system. 
DLT: https://github.com/borglab/gtsam/blob/c7e18e64e59b61d158318e469759420ca9c78f20/gtsam/geometry/triangulation.cpp#L85
LOST: https://github.com/borglab/gtsam/blob/c7e18e64e59b61d158318e469759420ca9c78f20/gtsam/geometry/triangulation.cpp#L124